E0392 needs to be updated to new format

From: 

E0392 needs a span_label, updating it from:

```
error[E0392]: parameter `T` is never used
  --> src/test/compile-fail/E0392.rs:11:10
   |
11 | enum Foo<T> { Bar } //~ ERROR E0392
   |          ^
   |
   = help: consider removing `T` or using a marker such as `std::marker::PhantomData`
```

To:

```
error[E0392]: parameter `T` is never used
  --> src/test/compile-fail/E0392.rs:11:10
   |
11 | enum Foo<T> { Bar } //~ ERROR E0392
   |          ^ unused type parameter
   |
   = help: consider removing `T` or using a marker such as `std::marker::PhantomData`
```

--- a/src/librustc_typeck/check/wfcheck.rs
+++ b/src/librustc_typeck/check/wfcheck.rs
@@ -653,8 +653,10 @@ fn error_380(ccx: &CrateCtxt, span: Span) {
 
 fn error_392<'a, 'tcx>(ccx: &CrateCtxt<'a, 'tcx>, span: Span, param_name: ast::Name)
                        -> DiagnosticBuilder<'tcx> {
-    struct_span_err!(ccx.tcx.sess, span, E0392,
-                     "parameter `{}` is never used", param_name)
+    let mut err = struct_span_err!(ccx.tcx.sess, span, E0392,
+                  "parameter `{}` is never used", param_name);
+    err.span_label(span, &format!("unused type parameter"));
+    err
 }
 
 fn error_194(tcx: TyCtxt, span: Span, name: ast::Name) {

--- a/src/test/compile-fail/E0392.rs
+++ b/src/test/compile-fail/E0392.rs
@@ -9,6 +9,7 @@
 // except according to those terms.
 
 enum Foo<T> { Bar } //~ ERROR E0392
+                    //~| NOTE unused type parameter
 
 fn main() {
 }

